src/analysis/inner_product_space/positive.lean

```diff
@@ -124,3 +126,198 @@ end
 end complex
 
 end continuous_linear_map
+
+namespace linear_map
+
+open linear_map
+
+variables {V : Type*} [inner_product_space ℂ V]
+
+local notation `e` := is_symmetric.eigenvector_basis
+open_locale big_operators
+
+/-- `T` is (semi-definite) positive if `∀ x : V, ⟪x, T x⟫_ℂ ≥ 0` -/
+def is_positive (T : V →ₗ[ℂ] V) :
+  Prop := ∀ x : V, 0 ≤ ⟪x, T x⟫_ℂ.re ∧ (⟪x, T x⟫_ℂ.re : ℂ) = ⟪x, T x⟫_ℂ
```

Comment:
The order on `ℂ` makes it so that nonnegative complex numbers exactly are nonnegative reals. So the following is a nicer way to phrase your condition.
```suggestion
def is_positive (T : V →ₗ[ℂ] V) : Prop := (0 : V → ℂ) ≤ λ x, ⟪x, T x⟫_ℂ
```

Reply:
that's giving me an error: `failed to synthesis type class instance for ... has_le (V → ℂ)`

Reply:
You need to do `open_locale complex_order` to enable this